main.tf

```diff
@@ -93,3 +93,11 @@ resource "google_kms_crypto_key_iam_binding" "encrypters" {
   members       = compact(split(",", var.encrypters[count.index]))
 }
 
+module "autokey" {
```

Comment:
Because we set `count` for this module, it won't create an extra resource unless user pass an input for `var.autokey_folder_number`. 

We can remove `autokey` call from the root module and make a total independent submodule, but if we do that, users will need to make a separate/dedicate call for the submodule if they want to use autokey. The way it is now, the root module is responsible to call the submodule based on the inputs passed.

There are some pros and cons in each approach, but I like the idea to turn the submodule independent. I'll implement the change so you can take another look!